javainterop/nextEntry.go

```diff
@@ -0,0 +1,29 @@
+package javainterop
+
+import (
+	"io"
+
+	"github.com/moov-io/ach"
+)
+
+type EntryDetail ach.EntryDetail
+
+type BatchHeader ach.BatchHeader
+
+type Iterator ach.Iterator
+
+type NextEntryResponse struct {
+	NextBatchHeader BatchHeader
+	NextEntryDetail EntryDetail
+	NextEntryError  error
+}
+
+func (i *Iterator) NextEntry(r io.Reader) NextEntryResponse {
+	batchHeader, entryFile, err := ach.NewIterator(r).NextEntry()
+
+	return NextEntryResponse{
+		NextBatchHeader: BatchHeader(*batchHeader),
+		NextEntryDetail: EntryDetail(*entryFile),
+		NextEntryError:  err,
+	}
```

Comment:
We will want to make the `Iterator` a struct holding a reference to the underlying `ach.Iterator` so that each call to `NextEntry()` will progress record by record. What you have now wouldn't quite work.

Reply:
Yes, this is true.